Optimizer/FcPreprocessor.hs

```diff
@@ -0,0 +1,36 @@
+module Optimizer.FcPreprocessor (mergeAppAbsOptProg) where
+
+import Optimizer.FcTypes
+
+mergeAppAbsOptProg :: FcOptProgram -> FcOptProgram
+mergeAppAbsOptProg (FcPgmDataDecl decl pgm) = FcPgmDataDecl decl (mergeAppAbsOptProg pgm)
+mergeAppAbsOptProg (FcPgmValDecl  bind pgm) = FcPgmValDecl (mergeAppAbsOptBind bind) (mergeAppAbsOptProg pgm)
+mergeAppAbsOptProg (FcPgmTerm tm) = FcPgmTerm (mergeAppAbsOptTm tm)
+
+-- | Merge applications and abstractions
```

Comment:
Add a simple example here. While this may be clear now, it might not be clear to someone reading this code in the future what you mean with merging.